Serialization/ArgsList.cs

```diff
@@ -50,16 +58,29 @@ static ArgsList()
 				typeof(Graphic).Assembly,
 			};
 
-			var componentSubTypes = assemblies.SelectMany(a => a.GetTypes())
-				.Where(t => t.IsSubclassOf(typeof(Component)));
-
-			foreach (var type in componentSubTypes)
+			foreach (var type in assemblies.SelectMany(a => a.GetTypes()))
 			{
-				// Make sure this type has not been added already 
-				// (it can happen, since some types are shipped in multiple assemblies)
-				if (!componentTypes.ContainsKey(type.FullName))
+				var fullTypeName = type.FullName;
+				if (type.IsSubclassOf(typeof(Component)))
 				{
-					componentTypes.Add(type.FullName, type);
+					if (!componentTypes.ContainsKey(fullTypeName))
```

Comment:
Couldn't you do `componentTypes[fullTypeName] = type;` instead of the `ContainsKey` check here?